core/arch/arch_exports.h

```diff
@@ -234,6 +234,7 @@ typedef struct _local_state_extended_t {
 #    define TLS_REG2_SLOT ((ushort)offsetof(spill_state_t, a2))
 #    define TLS_REG3_SLOT ((ushort)offsetof(spill_state_t, a3))
 #    define TLS_REG_STOLEN_SLOT ((ushort)offsetof(spill_state_t, reg_stolen))
+#    define TLS_REG_TP_SLOT os_get_app_tls_base_offset(TLS_REG_LIB)
```

Comment:
This define is used only in core/arch/riscv64/emit_utils.c, right? 

Let's follow the examples in core/arch/aarchxx/mangle.c, and use os_get_app_tls_base_offset(TLS_REG_LIB) in emit_utils.c there instead.

Otherwise, we are adding an unnecessary dependency on unix/os_export.h

Reply:
It will be used in several functions in riscv64/mangle.c for mangling the `tp` register.

Reply:
I think we should avoid adding the dependency on core/unix/os_exports.h.

Reply:
> I think we should avoid adding the dependency on core/unix/os_exports.h.

Sure, fixed in https://github.com/DynamoRIO/dynamorio/pull/6385/commits/2f6542391479336ee09214a4befd85e5a419b25f. Also could you explain why? I don't quite follow, but I'd like to avoid this kind of issue in future patches.

Reply:
os_get_app_tls_base_offset is defined in core/unix/os_exports.h, and implemented in core/unix/os.c.

By adding os_get_app_tls_base_offset here, all files with RISCV64 defined which include arch_exports.h have to include os_exports.h, and link with the lib including os.c.

While it's the case today, i.e. RISCV64 is used only with UNIX, it'll be hard to decouple later if the situation changes in the future.

Reply:
Thanks!